### PR TITLE
add mAP metric

### PR DESCRIPTION
**Pull request by yangtcai**

### CODE REVIEW

**Review by williamstark01 — CHANGES_REQUESTED** [review]
Looks good, did you implemented it yourself?

I see that TorchMetrics has an implementation of mAP could it have been used instead? Worth keeping the library in mind for other metrics we might use in the future.
https://torchmetrics.readthedocs.io/en/stable/detection/mean_average_precision.html
https://github.com/Lightning-AI/metrics/blob/master/src/torchmetrics/detection/mean_ap.py

File: train.py [hunk 1]
```diff
@@ -43,12 +45,13 @@ def train_one_epoch(
             loss_dict[k] * weight_dict[k] for k in loss_dict.keys() if k in weight_dict
         )
         losssum += losses.item()
-
+        mAPsum += mean_average_precision(outputs, targets, num_classes=5)
```
[the remark this answers is not in the document]
Comment (yangtcai) [reply train.py:48]: Yes, it should be better convert to a variable :D

**Comment by yangtcai** [comment]
It‘s a great suggestion, I will rewrite the training stage and this stage using PyTorch Lightning ;P

**Comment by yangtcai** [comment]
Related to #34 #35 #36
I have added more Fasta information into our training, I do a quick analysis, and this time we have about 2.5k repeat sequences that can be trained. Also, I need to apologize for the answer of #31, we can add more types to our train because we just select the chromosome 1 to 22 and chrX, chrY, so other types may stay in chromosomes we do not cover. 

I think saving logs and experiment files in configuration maybe is a good idea, it can help us to review what's changed our results. :D  An old saying, 好记性不如烂笔头(hǎo jì xìng bù rú làn bǐ tóu).
I haven't tested it on the cluster, but it can run on my laptop. ;P

**Review by williamstark01 — APPROVED** [review]
Great work! And keep the Chinese phrases coming, it's great to be learning more! :star2:

> I have added more Fasta information into our training, I do a quick analysis, and this time we have about 2.5k repeat sequences that can be trained.

Is 2.5k the number of segments with repeats or the number of repeats in the included chromosomes?

Similarly, what does the 8.9k number in #34 represent?

> Also, I need to apologize for the answer of #31, we can add more types to our train because we just select the chromosome 1 to 22 and chrX, chrY, so other types may stay in chromosomes we do not cover.

No worries, I think it's best to simply select the list of classification values using a filter with pandas so that we don't have to think which of these values are included in the current dataset and which are not. Take a look at how this is done in the statistics notebook:
https://github.com/yangtcai/Ensembl-Repeat-Identification/blob/main/dataset_statistics.py#L241

There are a few more arguments that can be converted to hyperparameters, adding a note in #35

closes #25, #36

**Comment by williamstark01** [comment]
Alright, the numbers make sense now!
